src/services/character/character.service.ts

```diff
@@ -1,31 +1,49 @@
-import _ from 'lodash'
-import { Document } from 'mongoose'
-import { ICharacter } from '../../types/character/character'
-import characters from '../../data/models/character-schema'
+import _ from "lodash";
+import mongoose from "mongoose";
+import { ICharacter } from "../../types/character/character";
 
 export class CharacterService {
-  private static _instance: CharacterService
+  private static _instance: CharacterService;
 
   public static getInstance(): CharacterService {
-    if(_.isNil(CharacterService._instance)) {
-      CharacterService._instance = new CharacterService()
+    if (_.isNil(CharacterService._instance)) {
+      CharacterService._instance = new CharacterService();
     }
-    return CharacterService._instance
+    return CharacterService._instance;
   }
 
-  public async getEntity(ownerId: string):Promise<Document | void> {
-    return await characters.findOne({ ownerId }, (err: Error, characterFound: Document): ICharacter => {        
-      if (characterFound) return characterFound.schema.obj
-    })
-    .catch((err): void => {
-      console.log(`Error: Character not found because: ${err}`)
-      return
-    })
+  public getEntity(_ownerId: string): Promise<ICharacter | never> {
+    return new Promise((resolve, reject): void => {
+      void mongoose
+        .model(`characterSchema`)
+        .findOne({ ownerId: _ownerId })
+        .then((foundCharacter: mongoose.Document | null): void => {
+          if (this._isEntity(foundCharacter)) {
+            console.log(`Character found with name: ${foundCharacter.name}`);
+
+            resolve(foundCharacter);
+          } else {
+            reject(
+              Error(`Couldn't find it with the provided owner: ${_ownerId}`)
+            );
+          }
+        });
+    });
+  }
+
+  public setEntity(_ownerId: string, _characterToInsert: ICharacter): void {
+    mongoose
```

Comment:
Comparing with the method above, this should be the same kind of code yet this is not.
Did you change something in the way you used mongoose?
Because it seems that now you are handling complete promises.
If yes, the stuff above with returning a `new Promise` is no longer the best option and here you should just return mongoose and change the return type to a Promise

Reply:
I see. I will do that.

Reply:
I see. I didn't do the same kind of code because I don't need a feedback from the method. To put it simply: I don't care when the "promise" is resolved unless there is an error. It's no use to wait for the promise to resolve before doing something else.

Reply:
@asmodee-bellanie and this is your error.
Not returning Promises when possible is an anti-pattern.
You should whenever you could return a Promise.

Reply:
Ok. I see. I will return a promise.